presto-spark-base/src/main/java/com/facebook/presto/spark/node/PrestoSparkInternalNodeManager.java

```diff
@@ -38,30 +38,34 @@
     public Set<InternalNode> getActiveConnectorNodes(ConnectorId connectorId)
     {
         // TODO: Hack to make TPCH connector work
-        return ImmutableSet.of(
-                new InternalNode("spark1", URI.create("http://127.0.0.1:60041"), NodeVersion.UNKNOWN, false),
-                new InternalNode("spark2", URI.create("http://127.0.0.1:60042"), NodeVersion.UNKNOWN, false),
-                new InternalNode("spark3", URI.create("http://127.0.0.1:60043"), NodeVersion.UNKNOWN, false),
-                new InternalNode("spark4", URI.create("http://127.0.0.1:60044"), NodeVersion.UNKNOWN, false));
+        return getAllNodes().getActiveNodes();
     }
 
     @Override
     public InternalNode getCurrentNode()
     {
         // TODO: Hack to make TPCH connector work
-        return new InternalNode("spark1", URI.create("http://127.0.0.1:1111"), NodeVersion.UNKNOWN, false);
+        return new InternalNode("spark1", URI.create("http://127.0.0.1:1111"), NodeVersion.UNKNOWN, true);
     }
 
     @Override
     public Set<InternalNode> getCoordinators()
     {
-        throw new UnsupportedOperationException();
+        return ImmutableSet.of(getCurrentNode());
     }
 
     @Override
     public AllNodes getAllNodes()
     {
-        throw new UnsupportedOperationException();
+        return new AllNodes(
```

Comment:
nit: maybe this should be a static field instead?

Reply:
It is not going to be called on production anyway. It's pretty much a hack just to make TPCH / System connectors happy. Hopefully we will refactor `presto-main` a little bit and decouple it from the node management.

Reply:
Extracted static constants